[CI] BlockAccountingTests testBytesRefVector failing

**Build Scans:**
- [elasticsearch-intake #39203 / part3](https://gradle-enterprise.elastic.co/s/jgypfugzy5oic)
- [elasticsearch-pull-request #143491 / part-3](https://gradle-enterprise.elastic.co/s/6iix7p7by3f7k)
- [elasticsearch-pull-request #143491 / part-3](https://gradle-enterprise.elastic.co/s/gt4ukktkxb7vy)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:compute:test" --tests "org.elasticsearch.compute.data.BlockAccountingTests.testBytesRefVector" -Dtests.seed=3A9A4B837DD2900 -Dtests.locale=yi-UA -Dtests.timezone=Etc/GMT+0 -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.compute.data.BlockAccountingTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testBytesRefVector'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: (a value equal to or greater than <258L> and a value less than or equal to <10000L>)
     but: a value equal to or greater than <258L> <257L> was less than <258L>
```

**Issue Reasons:**
- [main] 3 failures in test testBytesRefVector (0.9% fail rate in 329 executions)
- [main] 2 failures in step part-3 (0.9% fail rate in 214 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -265,9 +265,6 @@ tests:
 - class: org.elasticsearch.index.mapper.DateRangeDocValuesLoaderTests
   method: testEmptyRangesAppendsNull
   issue: https://github.com/elastic/elasticsearch/issues/148378
-- class: org.elasticsearch.compute.data.BlockAccountingTests
-  method: testBytesRefVector
-  issue: https://github.com/elastic/elasticsearch/issues/148481
 - class: org.elasticsearch.xpack.ml.integration.ForecastIT
   method: testOverflowToDisk
   issue: https://github.com/elastic/elasticsearch/issues/148606

--- a/server/src/main/java/org/elasticsearch/common/util/BytesRefArray.java
+++ b/server/src/main/java/org/elasticsearch/common/util/BytesRefArray.java
@@ -495,7 +495,7 @@ public final class BytesRefArray extends AbstractRefCounted implements Accountab
             boolean success = false;
             try {
                 if (initialSize < HALF_PAGE_SIZE) {
-                    final int bytesLength = Math.max(1, (int) initialSize);
+                    final int bytesLength = (int) initialSize;
                     bigArrays.adjustBreaker(bytesLength, false);
                     pages[0] = currentPage = new byte[bytesLength];
                     pageCount = 1;

